Fix CAS server and GEVER portal URLs

lukasgraf opened the pull request:
This PR introduces separate concepts for

- the base URL for the **GEVER cluster**
- the **CAS server** URL
- the **GEVER portal** URL

---

The **CAS server URL** is only constructed once (during setup) and used to configure the `ftw.casauth` PAS plugin. After that, the CAS server URL will always be retrieved from that plugin's property (for example to build the link for single-logout).

The **GEVER portal URL** however is constructed dynamically.

Both those URLs happen to be the same in our current setups, but they're always built using the **base URL of the GEVER cluster** as a base.

The GEVER cluster's base URL is determined through a heuristic: If the admin unit's `public_url` ends with the admin unit's ID, we're probably dealing with a multi-admin-unit setup, so we strip the admin unit ID from the end in order to get the correct base URL for the cluster.

deiferni commented:
Looks :+1: so far

deiferni commented on opengever/setup/upgrades/casauth_profiles/4600/actions.xml:
```diff
@@ -0,0 +1,21 @@
+
+  <object name="user" meta_type="CMF Action Category">
+    <object name="my-portal" meta_type="CMF Action" i18n:domain="opengever.setup" insert-before="logout">
+      <property name="title" i18n:translate="">My portal</property>
```
do you have to repeat all properties, even though they are not changed?

lukasgraf replied:
Good point. We don't, updated accordingly.

lukasgraf commented:
@deif this one is ready for merging as far as I'm concerned. Changes since your last look at it:
- Dropped redundant properties from upgrade step
- Added heuristic to determine correct cluster URL
- Added some more tests

deiferni commented on docs/HISTORY.txt:
```diff
@@ -5,7 +5,25 @@ Changelog
 
-- Nothing changed yet.
+- Cluster URL: Add heuristic to determine the correct cluster base URL
+  for both single- and multi-admin-unit setups.
```
please add an [lgraf] here, too :wink:

deiferni commented on opengever/base/casauth.py:
```diff
@@ -13,6 +12,50 @@ def is_cas_auth_enabled():
+    # dealing with a deployment with multiple admin units. So we strip the
+    # admin unit ID from the end of the URL to get the correct cluster base.
+    #
+    # This heuristic will later be replaced with a configuration of the$
```
the~~$~~

deiferni commented on opengever/base/tests/test_gever_state.py:
```diff
@@ -0,0 +1,29 @@
+
+    def setUp(self):
+        super(TestGeverStateView, self).setUp()
+        get_current_admin_unit().public_url = 'http://foo.org/cluster/fd'
```
as discussed, `fd` is confusing here, i'd suggest to remove it

lukasgraf commented:
@deiferni updated.
